django/db/models/fields/related.py

```diff
@@ -1375,6 +1376,27 @@ def _check_relationship_model(self, from_model=None, **kwargs):
 
         return errors
 
+    def _check_table_uniqueness(self, **kwargs):
+        if isinstance(self.remote_field.through, six.string_types):
+            return []
+        registered_tables = {
+            model._meta.db_table: model
+            for model in self.opts.apps.get_models(include_auto_created=True)
+            if model != self.remote_field.through
+        }
+        m2m_db_table = self.m2m_db_table()
+        if m2m_db_table in set(registered_tables.keys()):
+            msg = 'Field is using a table that has already been registered by \'%s.%s.%s\'.'
+            table = registered_tables[m2m_db_table]
```

Comment:
The `table` variable is actually a `models.Model` instance so it might be good to rename it to `model`.

In the case of auto-created models `model._meta.auto_created` will be pointing at the model at the origin of the creation else it will be `False`. When it's `False` the resulting message should be of the form `(opts.app_label, opts.object_name)` else it should be of the form `(opts.app_label, opts.object_name, field.name)` where `field` is retrieved from iterating over `model._meta.auto_created._meta.many_to_many` where `field.remote_field.through is model`.